components/p3a/brave_p3a_service.cc

```diff
@@ -176,7 +186,11 @@ void BraveP3AService::RegisterDynamicMetric(const std::string& histogram_name,
                                             MetricLogType log_type,
                                             bool should_be_on_ui_thread) {
   if (should_be_on_ui_thread) {
+#if BUILDFLAG(IS_IOS)
```

Comment:
can we add a function for this dchecks

Reply:
My worry with adding a function would be that the DCHECK's would execute in my function rather than at the body of this function.  Maybe the function could be inlinable to avoid that?